components/sidebar/sidebar_channel/index.js

```diff
@@ -48,11 +49,18 @@ function mapStateToProps(state, ownProps) {
         teammate = getUser(state, channel.teammate_id);
     }
 
+    let channelDisplayName = channel.display_name;
+    if (teammate && ownProps.currentUserId === teammate.id) {
+        channelDisplayName = displayUsername(ownProps.currentUserId);
+    } else if (teammate && channel.type === Constants.DM_CHANNEL) {
+        channelDisplayName = displayUsername(teammate.id);
+    }
+
```

Comment:
I think it would be easier to follow that we're doing this conditional if statements whenever the channel type is DM. So it can be like:
```
    let teammate = null;
    let channelTeammateId = null;
    let channelTeammateDeletedAt = null;
    let channelDisplayName = channel.display_name;
    if (channel.type === Constants.DM_CHANNEL) {
        if (ownProps.currentUserId === channel.teammate_id) {
            channelDisplayName = displayUsername(ownProps.currentUserId);
        } else {
            channelDisplayName = displayUsername(channel.teammate_id);
        }

        teammate = getUser(state, channel.teammate_id);
        if (teammate) {
            channelTeammateId = teammate.id;
            channelTeammateDeletedAt = teammate.delete_at;
        }
    }
```

then return 
```
return {
    ...
    channelTeammateId,
    channelTeammateDeletedAt,
    ...
}
```
What do you think?

Reply:
I like this suggestion, makes the logic cleaner. I'll change it. 👍